src/main/java/net/fabricmc/loom/mixin/MixinMappingWriterTiny.java

```diff
@@ -44,22 +50,59 @@ public MixinMappingWriterTiny(Messager messager, Filer filer) {
 	}
 
 	@Override
-	public void write(String output, ObfuscationType type, IMappingConsumer.MappingSet<MappingField> fields, IMappingConsumer.MappingSet<MappingMethod> methods) {
-		if (output != null) {
-			String from = type.getKey().split(":")[0];
-			String to = type.getKey().split(":")[1];
-
-			try (PrintWriter writer = this.openFileWriter(output, type + " output TinyMappings")) {
-				writer.println(String.format("v1\t%s\t%s", from, to));
-				for (IMappingConsumer.MappingSet.Pair<MappingField> pair : fields) {
-					writer.println(String.format("FIELD\t%s\t%s\t%s\t%s", pair.from.getOwner(), pair.from.getDesc(), pair.from.getSimpleName(), pair.to.getSimpleName()));
-				}
-				for (IMappingConsumer.MappingSet.Pair<MappingMethod> pair : methods) {
-					writer.println(String.format("METHOD\t%s\t%s\t%s\t%s", pair.from.getOwner(), pair.from.getDesc(), pair.from.getSimpleName(), pair.to.getSimpleName()));
+	public void write(String output, ObfuscationType type, IMappingConsumer.MappingSet<MappingField> fields,
+					  IMappingConsumer.MappingSet<MappingMethod> methods) {
+		if (output == null) {
+			return;
+		}
+		String[] parts = type.getKey().split(":");
+		String from = parts[0];
+		String to = parts[1];
+
+		Map<String, List<String>> classesData = new TreeMap<>();
+
+		print(classesData, "f", fields);
+		print(classesData, "m", methods);
+
+		try (PrintWriter writer = this.openFileWriter(output, type + " output TinyMappings")) {
+			writer.println(String.format("tiny\t2\t0\t%s\t%s", from, to));
```

Comment:
I think this should still distrubute v1 mappings here. I dont see any major reason to move to v2 here, doing this will require a loader update right?

Reply:
Hmm, isn't this only exposed to loom? let me check.

Reply:
You migh be right, so its not as much of an issue. I just dont see the point in breaking it when its not really needed?

Do you disagree?

Reply:
Let me look into loader first again. very likely the loader uses the data from refmap

Reply:
Turns out this is **only** supplied to the tiny remapper in the remap jar task; it has no other usage.

Reply:
Fixed.